framework/src/utils/Conversion.C

```diff
@@ -212,20 +215,31 @@ namespace Moose
   {
     switch (t)
     {
-    case EXEC_INITIAL:        return "INITIAL";
-    case EXEC_LINEAR:         return "LINEAR";
-    case EXEC_NONLINEAR:      return "NONLINEAR";
-    case EXEC_TIMESTEP_END:   return "TIMESTEP_END";
-    case EXEC_TIMESTEP_BEGIN: return "TIMESTEP_BEGIN";
-    case EXEC_CUSTOM:         return "CUSTOM";
-    case EXEC_FINAL:          return "FINAL";
-    case EXEC_FORCED:         return "FORCED";
-    case EXEC_FAILED:         return "FAILED";
-    case EXEC_SUBDOMAIN:      return "SUBDOMAIN";
-    case EXEC_NONE:           return "NONE";
+      case EXEC_INITIAL:        return "INITIAL";
+      case EXEC_LINEAR:         return "LINEAR";
+      case EXEC_NONLINEAR:      return "NONLINEAR";
+      case EXEC_TIMESTEP_END:   return "TIMESTEP_END";
+      case EXEC_TIMESTEP_BEGIN: return "TIMESTEP_BEGIN";
+      case EXEC_CUSTOM:         return "CUSTOM";
+      case EXEC_FINAL:          return "FINAL";
+      case EXEC_FORCED:         return "FORCED";
+      case EXEC_FAILED:         return "FAILED";
+      case EXEC_SUBDOMAIN:      return "SUBDOMAIN";
+      case EXEC_NONE:           return "NONE";
     }
     return "";
   }
+
+  template<>
+  std::string stringify(const Real & t)
+  {
+    std::ostringstream os;
+    // the correct value would be std::numeric_limits<Real>::max_digits10 (C++11)
+    os << std::scientific
+       << std::setprecision(std::numeric_limits<Real>::digits10 + 1)
```

Comment:
digits10+1?

"For an IEEE 754 float, digits10 is 6 and max_digits10 is 9; for a double it is 15 and 17. Note that both these numbers should not be confused with the actual decimal precision of floating-point numbers."

[link](http://stackoverflow.com/questions/22458355/what-is-the-purpose-of-max-digits10-and-how-is-it-different-from-digits10)